### ISSUE TITLE
Ambiguous class resolution

### ISSUE THREAD

**Issue by it-can** [thephpleague/flysystem-local#1]
just updated my Laravel 10 composer and got these warnings

```
- Upgrading league/flysystem (3.14.0 => 3.15.0)
  - Upgrading league/flysystem-aws-s3-v3 (3.13.0 => 3.15.0)
  - Upgrading league/flysystem-ftp (3.13.0 => 3.15.0)
  - Locking league/flysystem-local (3.15.0)
  - Upgrading league/flysystem-sftp-v3 (3.14.0 => 3.15.0)

Warning: Ambiguous class resolution, "League\Flysystem\Local\LocalFilesystemAdapter" was found in both "/vendor/league/flysystem-local/LocalFilesystemAdapter.php" and "/vendor/league/flysystem/src/Local/LocalFilesystemAdapter.php", the first will be used.
Warning: Ambiguous class resolution, "League\Flysystem\Local\FallbackMimeTypeDetector" was found in both "/vendor/league/flysystem-local/FallbackMimeTypeDetector.php" and "/vendor/league/flysystem/src/Local/FallbackMimeTypeDetector.php", the first will be used.
```

**Comment by it-can** [comment]
fixed by version 3.15.1

**Comment by some0ne2** [comment]
@it-can can you please explain how you fixed this?
Stumbled accross the same problem:

> 
> www-data-2@9342e499886e:~$ php composer.phar depends  league/flysystem-local
> league/flysystem 3.23.0 requires league/flysystem-local (^3.0.0) 
> www-data-2@9342e499886e:~$ php composer.phar depends league/flysystem
> laravel/framework      v10.40.0 requires league/flysystem (^3.8.0) 
> league/flysystem-local 3.23.0   requires league/flysystem (^3.0.0) 

Thanks in advance!
